crucible-mir-comp/src/Mir/Compositional/Override.hs

```diff
@@ -287,7 +287,12 @@ runSpec myCS mh ms = ovrWithBackend $ \bak ->
     let postAllocDefs = filter (\(k,_v) -> not $ Map.member k preAllocMap) $
             Map.toList $ ms ^. MS.csPostState . MS.csAllocs
     postAllocMap <- liftM Map.fromList $ forM postAllocDefs $ \(alloc, Some allocSpec) -> do
-        ref <- newMirRefSim (allocSpec ^. maType)
+        let vecRepr = MirVectorRepr (allocSpec ^. maType)
+        vecRef <- newMirRefSim vecRepr
+        writeMirRefSim vecRepr vecRef $
+            MirVector_PartialVector $ V.replicate (allocSpec ^. maLen) W4.Unassigned
+        zero <- liftIO $ W4.bvLit sym knownRepr $ BV.zero knownRepr
+        ref <- subindexMirRefSim (allocSpec ^. maType) vecRef zero
```

Comment:
It might be worth writing a brief Note that explains why we go through the effort of writing out a `MirVector_PartialVector` here instead of just returning a fresh reference. Once that Note is written, we could also cite it near [this very similar code in SAW](https://github.com/GaloisInc/saw-script/blob/0c7904230cd29e20b48c7440f419ebefe0548c75/saw-central/src/SAWCentral/Crucible/MIR/ResolveSetupValue.hs#L1600-L1610).